pennylane/devices/qubit/apply_operation.py

```diff
@@ -407,14 +407,84 @@ def apply_pauliz(op: qml.Z, state, is_state_batched: bool = False, debugger=None
     return math.stack([state[sl_0], state1], axis=axis)
 
 
+@apply_operation.register
+def apply_phaseshift(op: qml.PhaseShift, state, is_state_batched: bool = False, debugger=None, **_):
+    """Apply PhaseShift to state."""
+
+    params = op.parameters[0]
+    if is_state_batched or (op.batch_size is not None and len(params) > 1):
+        params = math.atleast_1d(params)
+        slices = []
+        for i, p in enumerate(params):
+            slices.append(
+                apply_phaseshift(
+                    qml.PhaseShift(p, wires=op.wires),
+                    state[i] if is_state_batched else state,
+                    is_state_batched=False,
+                )
+            )
+        return math.stack(slices, axis=0)
```

Comment:
Instead of looping, you can vectorize the computation quite efficiently using [this implementation](https://stackoverflow.com/a/62655664)